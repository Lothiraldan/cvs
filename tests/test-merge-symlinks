#!/bin/sh

"$TESTDIR/hghave" symlink || exit 80

cat > echo.py <<EOF
#!/usr/bin/env python
import os
for k in ('HG_MY_ISLINK', 'HG_OTHER_ISLINK', 'HG_BASE_ISLINK'):
    print k, os.environ[k]
EOF
chmod +x echo.py

# Create 2 heads containing the same file, once as
# a file, once as a link.
echo % create heads
hg init t
cd t
echo a > a
hg ci -qAm t0
echo l > l
hg ci -qAm t1
hg up -C 0
ln -s a l
hg ci -qAm t2

# Merge them and display *_ISLINK vars
echo % merge heads
HGMERGE=../echo.py hg merge
