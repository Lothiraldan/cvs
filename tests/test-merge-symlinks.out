% create heads
0 files updated, 0 files merged, 1 files removed, 0 files unresolved
% merge heads
HG_MY_ISLINK 1
HG_OTHER_ISLINK 0
HG_BASE_ISLINK 0
merging l
0 files updated, 1 files merged, 0 files removed, 0 files unresolved
(branch merge, don't forget to commit)
